tools/EdgeCases/Http/HttpClientEdgeCases.cs

```diff
@@ -0,0 +1,19 @@
+using System.Web;
+using Facility.Core;
+
+namespace EdgeCases.Http;
+
+public sealed partial class HttpClientEdgeCases
+{
+	protected override Task RequestReadyAsync(HttpRequestMessage httpRequest, ServiceDto requestDto, CancellationToken cancellationToken)
+	{
+		if (requestDto is CustomHttpRequestDto customDto)
+		{
+			const string prefix = "extra-";
+			var query = HttpUtility.ParseQueryString(httpRequest.RequestUri.Query);
+			customDto.Extras = query.AllKeys.Where(x => x?.StartsWith(prefix, StringComparison.Ordinal) is true).ToDictionary(x => x.Substring(prefix.Length), x => query[x]);
```

Comment:
If I'm understanding correctly, I think I'd want to be doing the opposite here: adding query string parameters from the values given in `CustomHttpRequestDto.Extras`.

Reply:
Ha, good point! We'd need this logic on the server side... I'll fix it!

Reply:
Fixed!